tests/framework/clients/helm/helm.go

```diff
@@ -0,0 +1,150 @@
+package helm
+
+import (
+	"context"
+	"os/exec"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var helmCmd = "helm_v3"
+
+func InstallRancher(restConfig *rest.Config) error {
+	//  ClientSet of kubernetes
+	clientset, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	// Create namespace cattle-system
+	namespace := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "cattle-system"}}
+	_, err = clientset.CoreV1().Namespaces().Create(context.Background(), namespace, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Install cert-manager chart
+	err = InstallCertManager(restConfig)
+	if err != nil {
+		return err
+	}
+
+	// Add Rancher helm repo
+	err = AddHelmRepo("rancher-stable", "https://releases.rancher.com/server-charts/stable")
+	if err != nil {
+		return err
+	}
+
+	// Install Rancher Chart
+	err = InstallChart("rancher",
+		"rancher-stable/rancher",
+		"cattle-system",
+		"",
+		"--set",
+		"hostname=localhost",
+		"--set",
+		"bootstrapPassword=admin",
+		"--set",
+		"useBundledSystemChart=true",
+		"--set",
+		"replicas=1")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func InstallCertManager(restConfig *rest.Config) error {
+	//  ClientSet of kubernetes
+	clientset, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	// Create namespace cert-manager
+	namespace := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "cert-manager"}}
+	_, err = clientset.CoreV1().Namespaces().Create(context.Background(), namespace, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Add cert-manager Helm Repo
+	err = AddHelmRepo("jetstack", "https://charts.jetstack.io")
+	if err != nil {
+		return err
+	}
+
+	// Install cert-manager Chart
+	err = InstallChart("cert-manager", "jetstack/cert-manager", "cert-manager", "", "--set", "installCRDs=true")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Install a helm chart
+func InstallChart(releaseName, helmRepo, namespace, version string, args ...string) error {
```

Comment:
I am sure I am missing context about it, and the answer probably depends on the plan and how the container is going to be deleted; would it make sense to add a session to bind an uninstall chart action in this function? 

Sessions allow us to register different functions -- they are usually used for delete/uninstall calls. We can think about them as queues. The session cleanup can be called either in teardown or at the end of the tests, wherever we see the best fit.

Reply:
Something like this Rohit: https://github.com/rancher/rancher/blob/4367bc51f4de4c59250e7b20ddf2a1382dab04e4/tests/framework/clients/k3d/k3d.go#L34-L36

Reply:
Added the registerCleanup for both Install and Upgrade charts.